### DISCUSSION TITLE
Possible to read content without decoding it?

### DISCUSSION THREAD

**Discussion by pmeier** [encode/httpx#2900] [Q&A] [answered]
IIUC

```python
httpx.Response().read()
```

always decodes the content before reading in case it is able to:

https://github.com/encode/httpx/blob/e63b6594f2863b7c8274eb0991ebc6cad63661f7/httpx/_models.py#L830-L834

Is it possible to get the raw content, or rather `await response.aread()` it, without applying any decoding?

**Comment by karpetrosyan** [comment, marked as answer]
You can use iter_raw directly

Like so:

```python
import httpx

with httpx.stream("GET", "https://www.youtube.com") as response:
    print(response.headers['content-encoding'])  # gzip
    raw_content = b''.join(response.iter_raw())  # gziped content
```

**Reply by pmeier** [reply]
:facepalm: Thanks, that is exactly what I'm looking for. Not sure how I missed it in the documentation.
